File: Arrays/1. Find pair with given sum in the array/python/BruteForce.py
# arr = input("arr: ")
# sum = input("sum: ")
arr = '8 7 2 5 3 1'
sum = '10'
sum = int(sum)
arr = list(map(int,arr.split(" ")))

print('arr: ', arr, type(arr))
print('sum: ', sum, type(sum))

def findPair(arr, sum):
    for i in arr:
        for j in arr:
            if (i==j):
                continue
            elif (i+j == sum):
                print(i,j)

findPair(arr,sum)

File: Arrays/1. Find pair with given sum in the array/c++/BruteForce.cpp
#include <iostream>
#include <algorithm>
#include <utility>
#include <vector>
#include <sstream>

using namespace std;

void printPair(pair<int, int> pair)
{
    cout << "(" << pair.first << ", " << pair.second << ")" << endl;
}

void findPair(vector<int> arr, int sum)
{
    bool not_found = true;
    for (int i = 0; i < arr.size(); i++)
    {
        for (int j = 0; j < arr.size(); j++)
        {
            if (i == j)
            {
                continue;
            }

            else if (arr[i] + arr[j] == sum)
            {
                printPair(pair<int, int>(arr[i], arr[j]));
                not_found = false;
            }
        }
    }

    if (not_found)
    {
        cout << "No pairs found!" << endl;
    }
}

int main()
{
    string input;
    getline(cin, input);
    stringstream ss(input);
    string temp;

    vector<int> arr;

    while (getline(ss, temp, ' '))
    {
        arr.push_back(stoi(temp));
    }

    int sum;
    scanf("%d", &sum);

    // input example = 8 7 2 5 3 1 ; sum = 10

    findPair(arr, sum);

    return 0;
}